### ISSUE TITLE
Offer a list of all streams, and a way to subscribe

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#188]
Sometimes one wants to navigate to a stream when not already subscribed, or to subscribe to a stream.

In zulip-mobile this is offered on the "All streams" screen, reached from the bottom of the "Streams" tab.

We should offer equivalent functionality, whether through a similar design or something different.

**Comment by gnprice** [comment]
Yes indeed — this screen should show all streams. (Well, all streams that the user is able to see.)

Cross-referencing, we have a more detailed discussion in this chat thread:
https://chat.zulip.org/#narrow/stream/48-mobile/topic/all-streams.20screen.20.23F188/near/1761243

**Comment by gnprice** [comment]
This feature now has a design in Figma (thanks Vlad!). I've updated the description to reflect that.

**Comment by chrisbobbe** [comment]
When this is done we should update the Help Center doc. We removed instructions for the legacy app when preparing to launch the new app: zulip/zulip@977c3943d198d1f9482f9c7f695e1f356fd94c98

**Comment by alya** [comment]
Requested [in the FHIR community](https://chat.fhir.org/#narrow/channel/179238-Zulip/topic/search.20in.20stream.3F/near/526273818).

**Comment by gnprice** [comment]
We had a couple of requests for this feature today:
* [#mobile > ✔ flutter: browse channels](https://chat.zulip.org/#narrow/channel/48-mobile/topic/.E2.9C.94.20flutter.3A.20browse.20channels/with/2226839)
* #1733

**Comment by alya** [comment]
Another request in the Rust community: [#zulip > new mobile app feedback @ 💬](https://rust-lang.zulipchat.com/#narrow/channel/122653-zulip/topic/new.20mobile.20app.20feedback/near/531274313)

**Comment by chrisbobbe** [comment]
Another request for this on CZO: [#mobile-design > Mobile app channel list @ 💬](https://chat.zulip.org/#narrow/channel/530-mobile-design/topic/Mobile.20app.20channel.20list/near/2230749)
